testdata/sqllogictests/functions/hashing.slt

```diff
@@ -106,3 +106,62 @@ query I
 select siphash('42');
 ----
 8771948186893062792
+
+########################################################################
+#
+# partition_results(<value>, <num_shards>, <shard_id>)
+#
+########################################################################
+
+statement error
+select partition_results();
+
+statement error
+select partition_results('buddy', 100, 2, 3);
+
+statement error
+select partition_results('buddy', -100, -2);
+
+statement error
+select partition_results('buddy', 100, 200);
+
+statement error
+select partition_results('9001', '100', '10');
+
+statement error
+select partition_results(9001, 100, '10');
+
+statement error
+select partition_results(9001, '100', 10);
+
+statement ok
+select partition_results(100, 10, 0);
+
+statement ok
+select partition_results(100, 10.0, 1.0);
+
+statement error
+select partition_results(100, 10.5, 1.5);
+
+statement error
+select partition_results(16, 4, 4);
+
+query B
```

Comment:
Technically there is no 'B'. Only 'I', 'R', and 'T'. For bools, I just use 'T'. I also don't think the types are actually checked, just that there's the correct number of them.

https://github.com/risinglightdb/sqllogictest-rs/blob/main/sqllogictest/src/column_type.rs#L11-L43